docs/api/ShallowWrapper/instance.md

```diff
@@ -2,20 +2,62 @@
 
 Gets the instance of the component being rendered as the root node passed into `shallow()`.
 
-NOTE: can only be called on a wrapper instance that is also the root instance.
+NOTE: can only be called on a wrapper instance that is also the root instance. With React `16.x`, `instance()` returns `null` for stateless React component/stateless functional components. See example:
 
+#### Returns (React 16.x)
 
+- `ReactComponent`: The stateful React component instance.
+- `null`: If stateless React component was wrapped.
 
-#### Returns
+#### Returns (React 15.x)
 
-`ReactComponent`: The component instance.
+- `ReactComponent`: The component instance.
 
+#### Example
+
+#### Preconditions
+<!-- eslint react/prop-types: 0, react/prefer-stateless-function: 0 -->
+```jsx
+function Stateless() {
+  return <div>Stateless</div>;
+}
+
+class Stateful extends React.Component {
+  render() {
+    return <div>Stateful</div>;
+  }
+}
+```
+#### React 16.x
+```jsx
+test('shallow wrapper instance should not be null', () => {
+  const wrapper = shallow(<Stateless />);
+  const instance = wrapper.instance();
 
+  expect(instance).not.toBe(null); // This will fail, because instance is null
+});
 
-#### Example
+test('shallow wrapper instance should not be null', () => {
+  const wrapper = shallow(<Stateful />);
+  const instance = wrapper.instance();
 
+  expect(instance).not.toBe(null); // This will pass
+});
+```
+#### React 15.x
 ```jsx
-const wrapper = shallow(<MyComponent />);
-const inst = wrapper.instance();
-expect(inst).to.be.instanceOf(MyComponent);
+test('shallow wrapper instance should not be null', () => {
+  const wrapper = shallow(<Stateless />);
+  const instance = wrapper.instance();
+
+  expect(instance).not.toBe(null); // This will pass
+});
+
+test('shallow wrapper instance should not be null', () => {
+  const wrapper = shallow(<Stateful />);
+  const instance = wrapper.instance();
+
+  expect(instance).not.toBe(null); // This will pass
```

Comment:
let's preserve the original assertion here too: `expect(inst).to.be.instanceOf(MyComponent);`

Reply:
From original example it's not clear (as for me) what is `< MyComponent />`. I mean, is it stateful or stateless. And whole purpose of this update is to show this specific case.

Reply:
It's fine to show "null or not null"; I'm asking not to delete the "instance of MyComponent" checks.

Reply:
Sorry, I did not catch it.